app/controllers/cancelRegistration/CancellationRequestDoneController.scala

```diff
@@ -37,40 +39,47 @@ class CancellationRequestDoneController @Inject()(
-      val formattedTime = getSentDateTime.format(timeFormatter)
 
-      val returnPeriodFormat = DateTimeFormatter.ofPattern("MMMM yyyy")
-      val nextReturnPeriod = ReturnPeriod(getSentDateTime.toLocalDate)
-      val returnPeriodStart = nextReturnPeriod.start.format(returnPeriodFormat)
-      val returnPeriodEnd = nextReturnPeriod.end.format(returnPeriodFormat)
+      request.userAnswers.submittedOn match {
+        case Some(submittedOnDate) =>
+          val getSentDateTime = LocalDateTime.ofInstant(submittedOnDate, ZoneId.of("Europe/London"))
+          val formattedDate = getSentDateTime.format(DateTimeFormatter.ofPattern("d MMMM yyyy"))
+          val formattedTime = getSentDateTime.format(DateTimeFormatter.ofPattern("h:mma"))
 
-      val deadlineStartFormat = DateTimeFormatter.ofPattern("d MMMM")
-      val deadlineStart = nextReturnPeriod.end.plusDays(1).format(deadlineStartFormat)
+          val returnPeriodFormat = DateTimeFormatter.ofPattern("MMMM yyyy")
+          val nextReturnPeriod = ReturnPeriod(getSentDateTime.toLocalDate)
+          val returnPeriodStart = nextReturnPeriod.start.format(returnPeriodFormat)
+          val returnPeriodEnd = nextReturnPeriod.end.format(returnPeriodFormat)
 
-      val deadlineEndFormat = DateTimeFormatter.ofPattern("d MMMM yyyy")
-      val deadlineEnd = nextReturnPeriod.deadline.format(deadlineEndFormat)
+          val deadlineStartFormat = DateTimeFormatter.ofPattern("d MMMM")
+          val deadlineStart = nextReturnPeriod.end.plusDays(1).format(deadlineStartFormat)
 
-      val cancelRegistrationSummary: (String, SummaryList) = ("", SummaryListViewModel(
-        rows = Seq(
-          ReasonSummary.row(request.userAnswers, isCheckAnswers = false),
-          CancelRegistrationDateSummary.row(request.userAnswers, isCheckAnswers = false)
-        ))
-      )
+          val deadlineEndFormat = DateTimeFormatter.ofPattern("d MMMM yyyy")
+          val deadlineEnd = nextReturnPeriod.deadline.format(deadlineEndFormat)
 
-      Ok(view(
-        formattedDate, formattedTime,
-        returnPeriodStart, returnPeriodEnd,
-        deadlineStart, deadlineEnd,
-        request.subscription.orgName, Seq(cancelRegistrationSummary)))
+          val cancelRegistrationSummary: (String, SummaryList) = ("", SummaryListViewModel(
+            rows = Seq(
+              ReasonSummary.row(request.userAnswers, isCheckAnswers = false),
+              CancelRegistrationDateSummary.row(request.userAnswers, isCheckAnswers = false)
+            ))
+          )
+
+          Ok(view(
+            formattedDate, formattedTime,
+            returnPeriodStart, returnPeriodEnd,
+            deadlineStart, deadlineEnd,
+            request.subscription.orgName, Seq(cancelRegistrationSummary)))
+
+        case None => genericLogger.logger.error(s"[SoftDrinksIndustryLevyService [submitVariation] - unexpected response while attempting to retreive userAnswers submittedOnDate")
```

Comment:
The None on SubmittedOn means it is empty.  Should there be an Internal server error here or should the user go back to CYA?

Reply:
This question would also be the same for the other 3 done/sent controllers

Reply:
Underneath is the internal server error for this case